types/date-fp/package.json

```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "dependencies": {
+        "ts-toolbelt": "^6.3.3"
+    }
+}
```

Comment:
Hmm, I'm not a fan of adding a dependency just for the `Function.Curry` types. 

Since it's just for types, could you add it to `devDependencies` instead? Then I suppose my objections would be minimized. 

Alternatively, you can provide your own finite set of `Curry` overloads. It looks like each curried function takes at most 3 arguments, if so this code may be drop-in useable (although I have no experience comparing it against `ts-toolbelt` so YMMV)

https://github.com/strong-roots-capital/od/blob/master/src/curry.ts